pipeline.py

```diff
@@ -72,31 +93,124 @@ class TextProcessingPipeline:
     """
 
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        articles = self.corpus_manager.get_articles().values()
+
+        for article in articles:
+            raw_text = article.get_raw_text()
+            processed_tokens = self._process(raw_text)
+
+            cleaned_tokens = []
+            single_tagged_tokens = []
+            multiple_tagged_tokens = []
+
+            for processed_token in processed_tokens:
+                cleaned_tokens.append(processed_token.get_cleaned())
+                single_tagged_tokens.append(processed_token.get_single_tagged())
+                multiple_tagged_tokens.append(processed_token.get_multiple_tagged())
+
+            article.save_as(' '.join(cleaned_tokens), ArtifactType.cleaned)
+            article.save_as(' '.join(single_tagged_tokens), ArtifactType.single_tagged)
+            article.save_as(' '.join(multiple_tagged_tokens), ArtifactType.multiple_tagged)
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        # txt from pdf comes with words like след-ующий
+        # this replace deals with them
+        text = raw_text.replace('-\n', '')
+
+        result = Mystem().analyze(text)
+
+        morph_tokens = []
+        morph = pymorphy2.MorphAnalyzer()
+
+        for token in result:
+
+            # pre requisites for the token to be usable
+            if "analysis" not in token:
+                continue
+            if not token["analysis"]:
+                continue
+
+            original_word = token["text"]
+            if not re.match(r"[A-Za-zА-Яа-яЁё]", original_word):
+                continue
+
+            morph_token = MorphologicalToken(original_word=original_word)
 
+            # next pre requisite
+            if "lex" or "gr" not in token['analysis'][0]:
+                continue
+
+            # mystem tags
+            morph_token.normalized_form = token['analysis'][0]['lex']
+            morph_token.tags_mystem = token['analysis'][0]['gr']
+
+            # pymorphy tags
+            one_word = morph.parse(original_word)[0]
```

Comment:
no checks for pymorphy results, but it might be also producing some unexpected reports

Reply:
Sure, but I use already the word that has been checked through, I already take the word, rather than something empty. Pymystem checks are here because it doesn't produce one word, but rather a list of dictionaries, which are sometimes empty.